test/e2e_node/plugins/gcp-credential-provider/README.md

```diff
@@ -0,0 +1,35 @@
+# GCP credential provider for e2e testing
+
+This package contains a barebones implementation of the [kubelet GCP credential
+provider](https://github.com/kubernetes/cloud-provider-gcp/tree/master/cmd/auth-provider-gcp)
+for testing purposes only. This plugin SHOULD NOT be used in production.
+
+This credential provider is installed and configured in the node e2e tests by:
+
+1. Building the gcp-credential-provider binary and including it in the test archive
+   uploaded to the GCE remote node.
+
+2. Writing the credential provider config into the temporary workspace consumed
+  by the kubelet. The contents of the config should be something like this:
+
+```yaml
+kind: CredentialProviderConfig
+apiVersion: kubelet.config.k8s.io/v1alpha1
+providers:
+  - name: gcp-credential-provider
+    apiVersion: credentialprovider.kubelet.k8s.io/v1alpha1
+    matchImages:
```

Comment:
with this https://github.com/kubernetes/k8s.io/issues/3411 we may need to include `registry.k8s.io`. Not sure if private registry will also be moved there.

Reply:
ack -- I don't think any of the authenticated images use that but we should switch over when we do

Reply:
since it's for tests only, should it be just `*`?

Reply:
Hmm that would mean we run the auth flow for all images, not just the ones the plugin can handle right?

Btw, the match images are taken from [here](https://github.com/kubernetes/cloud-provider-gcp/blob/cf3a0e923f1d8992323edd8d29e3844341b6f949/cluster/gce/gci/configure-helper.sh#L3201-L3204)

Reply:
I don't know of any places where we don't use gcr.io for private images yet, but in the future it would be unexpected to run this plugin when the image is from another source

Reply:
I was mostly thinking from the perspective of not overcomplicating code as test knows which images will be pulled.

Practically `registry.k8s.io` will be just a proxy to gcr.io when running in GCE. Speculating here... it is interesting,  if this proxy will return `302` with the gcr.io address, will credential provider be used to pull from that redirected source?

Reply:
Hmm that's a good point -- if registry.k8s.io is just an alias for gcr.io domain we might need to adjust the plugin config to handle it